Add Windows packaging to Github Actions

vadi2 opened the pull request:
<!-- Keep the title short & concise so your mom can understand it -->
#### Brief overview of PR changes/additions
Add windows packaging to Github Actions
#### Motivation for adding to Mudlet
So we can get a 64bit Mudlet out via Github Actions - which gives us more concurrent builds - hopefully much quicker than in Appveyor
#### Other info (issues closed, discussion etc)
* We were linking Mudlet against YAJL, but that was not actually necessary - yajl is only used indirectly through Lua.
* Drive is no longer hardcoded to `C:\` and now can be changed to another one. This isn't reflected in the manual build script yet.
* Zziplib now uses cmake, autotools is heading for depreciation by the project

Background info on translating appveyor variables to github ones:
https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables
https://www.appveyor.com/docs/environment-variables/

SlySven commented on CI/after_success.ps1:
```diff
@@ -1,10 +1,15 @@
+tree /f $Script:BuildFolder
+
+Set-Location "$Script:BuildFolder\src\release"
 windeployqt.exe --release mudlet.exe
```
You may need to review this line after Qt 5.**14** as the `--release` option does not work - it seems to be assumed unless the *opposite* `--debug` option is given. https://bugreports.qt.io/browse/QTBUG-80806 may be relevant for this. Also `--no-virtualkeyboard` might be helpful as well, if we are not including whatever libraries *that* requires...

vadi2 replied:
Thanks, good to know

vadi2 committed:
Correct regex to use

vadi2 committed:
Try with a longer version string

vadi2 committed:
Export QT_BASE_DIR as needed

vadi2 committed:
Rename back to appveyor.set-build-info for diffing

vadi2 committed:
Remove autogenerated file

vadi2 committed:
Set variable outFile

vadi2 committed:
Move setting QT_BASE_DIR into function

vadi2 committed:
Move MINGW_BASE_DIR into function as well

vadi2 committed:
Improve upload message

vadi2 committed:
Add lastexitcodes spam

vadi2 committed:
Only set -github for github

vadi2 committed:
Set PR_NUMBER

vadi2 committed:
Remove comments

vadi2 committed:
Fix check for windows upload

vadi2 committed:
Add some more lastexitcode's debug

vadi2 committed:
Don't copy Lua 5.1 for now, doesn't work in github

vadi2 committed:
Set PR_NUMBER correctly

vadi2 committed:
Set the PR number correctly

vadi2 committed:
Append Github version right

vadi2 committed:
Don't double .zip.zip

vadi2 committed:
Remove Get-ChildItem env:

vadi2 committed:
Remove debug trace

vadi2 committed:
Remove lastexitcode checks

vadi2 committed:
Show D:\a\Mudlet recursively

vadi2 committed:
dlls

vadi2 committed:
Get libgcc_s_seh from QT_BASE_DIR

vadi2 committed:
Copy in libstdc++-6.dll

vadi2 committed:
Copy in libwinpthread-1.dll

SlySven commented:
As you are building a 64-bit Mudlet Windows build in GH Actions - what are the 32-bit Window users going to use? Remember they cannot run the 64-bit one - and, unless you tweak the `dblsqd push` lines in what is now called `./CI/after_success.ps1`, they are going to get the 64-bit one sent to them when the update is done... 💥

In a similar manner you MUST change the `<id>` field in the `mudlet.nuspec` file to reflect that it is a different product and - since you cannot use any form of braces (that I have found) and that the id must NOT end in a number - I was going to suggest (and that was what I was intending to use in *my* system) that the `<id>` should be `Mudlet_64_` for the release and `Mudlet_64_-PublicTestBuild` for the Public Test builds... You will probably also want to edit the `<title>` and `<iconUrl>` elements as well...

FYI The 64-bit OpenSSL libraries (licenced remember, like the 32-bit ones, under the **[dual OpenSSL and SSLeay license](Nohttps://www.openssl.org/source/license-openssl-ssleay.txt)** which is NOT compatible with GPL 2 or 3 licensed products UNLESS the latter has an additional exemption for all parts that link to the OpenSSL libraries) are:
32-bit|64-bit
----|---
`(32-bit path)/bin/libcrypto-1_1.dll`|`(64-bit path)/bin/libcrypto-1_1-x64.dll`
`(32-bit path)/bin/libssl-1_1.dll`|`(64-bit path)/bin/libssl-1_1-x64.dll`

From my experiments in 64/32 bit Windows builds you may also want the following extra libraries which Qt can make use of if it can find them dynamically:
* libjasper-4 to libwebpdemux-2 are additional image format handlers that Qt can use if they are present:
  *  `(32 or 64-bit path)/bin/libjasper-4.dll`
  *  `(32 or 64-bit path)/bin/libjpeg-8.dll`
  *  `(32 or 64-bit path)/bin/libtiff-5.dll`
  *  `(32 or 64-bit path)/bin/libwebp-7.dll`
  *  `(32 or 64-bit path)/bin/libwebpdemux-2.dll`
* SDL2 helps with Gamepad support that QtGamepad can use if it is present:
  *  `(32 or 64-bit path)/bin/SDL2.dll`

vadi2 commented:
Thanks, will look. As far as 32bit/64bit, I'll update the technical identifiers, though it should be pretty invisible to users - except on the one part where it matters, on the website. We should default to 64bit once it works, and offer 32bit as an option for those who really need it.

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

vadi2 committed:
Update comment for Lua DLLs

vadi2 committed:
Try Libzip 1.7.3 instead

vadi2 committed:
Copy in dictionaries

vadi2 committed:
Install LuaRocks

vadi2 committed:
Disable Luarocks for now

vadi2 committed:
Disable unneeded libraries and go back to original libzip

vadi2 committed:
Revert "Disable Luarocks for now"

This reverts commit 34ca73b6bfcd34461d523ef3a35be68d7f96ece0.

vadi2 committed:
Update Luarocks version

vadi2 committed:
Remove temp debug

vadi2 committed:
Debug show VCPKG_ROOT

vadi2 committed:
Add location parameter to CheckAndInstallLuarocks

vadi2 committed:
Plug in custom LuaRocks location

vadi2 committed:
Don't install quietly, see what's going wrong

vadi2 committed:
Plug in custom Lua dir for LuaRocks install

vadi2 committed:
Print Lua version

vadi2 committed:
Call Lua correctly

vadi2 committed:
Try checking output again

vadi2 committed:
Show $env:VCPKG_ROOT\installed\x64-mingw-dynamic\tools\lua

vadi2 committed:
Upload built lua.exe

vadi2 committed:
Remove last of lastexitcode

vadi2 committed:
Clean-up outFile after reading it

vadi2 committed:
Temporarily disable rest of build to get Lua artifact

vadi2 committed:
Re-enable rest of workflow

vadi2 committed:
Use vcpkg-provided Lua

vadi2 committed:
Revert "Use vcpkg-provided Lua"

This reverts commit da896c619d896dbf7aaffc5e3327dcc0febda5f5.

vadi2 committed:
Install Lua ourselves

vadi2 committed:
Set LuaRocks location explicitly

vadi2 committed:
Debug LFS install

vadi2 committed:
Try calling luarocks.bat explicitly

vadi2 committed:
Try exec

vadi2 committed:
Try absolute path

vadi2 committed:
Check that Lua works

vadi2 committed:
Try &, don't supress output

vadi2 committed:
Capitalise 'running cmake' like other steps

vadi2 committed:
Try just command alone

vadi2 committed:
Try luarocks --help bare

vadi2 committed:
Try without quotes

vadi2 committed:
Try calling just via Lua

vadi2 committed:
Back to working Luarocks 3.1.2

vadi2 committed:
Remove debug rubbish

vadi2 committed:
Install sqlite

vadi2 committed:
Remove Lua upload debug step

vadi2 committed:
Show VCPKG_ROOT root

vadi2 committed:
Build it first

vadi2 committed:
Supply path in github build

vadi2 committed:
Update it for yajl as well

vadi2 committed:
Re-add missing cd

vadi2 committed:
Print VCPKG_ROOT to find yajl_parse.h

vadi2 committed:
Debug yajl more

vadi2 committed:
Try forward slashes in yajl

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

vadi2 committed:
Try a fake path

vadi2 committed:
Try luarocks 3.3.1 [skip ci]

vadi2 committed:
Try with --verbose

vadi2 committed:
Print logfile after every run

vadi2 committed:
Try without prefix

vadi2 committed:
Try both variants

vadi2 committed:
Try adding LUA_INCDIR

vadi2 committed:
Back to backslashes

vadi2 committed:
Print root for luazip

vadi2 committed:
Run CheckAndInstallZziplib

vadi2 committed:
Print contents of CheckAndInstallZziplib

vadi2 committed:
Show makefile

vadi2 committed:
Try cmake variant

vadi2 committed:
Don't go too deep in

vadi2 committed:
Show what's there

vadi2 committed:
Try the original download

vadi2 committed:
Don't go too deep

vadi2 committed:
Inside zziplib-master

vadi2 committed:
Pring zzlib cmakeoutput

vadi2 committed:
Try building with Ninja

vadi2 committed:
Back to MinGW Makefiles

vadi2 committed:
Add --target libzzip

vadi2 committed:
Fix --target syntax

vadi2 committed:
Disable zziplib docs and tests

vadi2 committed:
Pass --build

vadi2 committed:
Add Zziplib to vcpkg dependencies

vadi2 committed:
Show zlib debug

vadi2 committed:
Get zlib build log right

vadi2 committed:
Back to our own zziplib, try with ninja

vadi2 committed:
Back to mingw makefiles

vadi2 committed:
Test for VCPKG_ROOT explicitly where needed

vadi2 committed:
A few more env fixes

vadi2 committed:
A few more env fixes #2

vadi2 committed:
Remove noshpath for zziplib

vadi2 committed:
Print logfile for libzip install

vadi2 committed:
Print logfile for right thing

vadi2 committed:
Ignore lua-zip for now

vadi2 committed:
Copy in LuaRocks we have

vadi2 committed:
Correct location for dictionaries

vadi2 committed:
Remove debug log print

vadi2 committed:
Add caching of nonvcpkg dependencies

vadi2 committed:
Don't error when folder already exists

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

vadi2 committed:
Disable vcpkg Lua, doesn't make working lua.exe

vadi2 committed:
Take out #

vadi2 committed:
Show $Env:MINGW_BASE_DIR

vadi2 committed:
Revert yajl to usual installation

vadi2 committed:
Provide LUA_INCDIR for Luarocks as well

vadi2 committed:
Set LUA_INCDIR to where Lua is

vadi2 committed:
Try both locations for LUA_INCDIR

vadi2 committed:
Install YAJL manually, so headers are found

vadi2 committed:
Clean up LuaYajl install

vadi2 committed:
Debug finding yajl

vadi2 committed:
Take out finding YAJL. I don't think we link to it?

vadi2 committed:
Copy libyajl from mingw installation path

vadi2 committed:
Copy in Lua as well

vadi2 committed:
Try to get LuaZip working

vadi2 committed:
Print log on failure instead of uploading to transfer.sh

vadi2 committed:
Try and debug Luazip

vadi2 committed:
Try cat with powershell

vadi2 committed:
Restore install order

vadi2 committed:
Upload log to appveyor instead

vadi2 committed:
Change linker lib name

vadi2 committed:
Remove debug luazip

vadi2 committed:
Remove debug finding yajl

vadi2 committed:
Print $Env:MINGW_BASE_DIR\lib\lua\5.1

vadi2 committed:
Cleanup

vadi2 committed:
Always show logs

vadi2 committed:
Print logfile after zziplib install

vadi2 committed:
Try to find libzzip-0.dll

vadi2 committed:
Copy in libzzip-0.dll to install zip as well

vadi2 committed:
Manually install libzzip-0.dll

vadi2 committed:
Remove debug print

vadi2 committed:
Minor cleanup

vadi2 committed:
Build with same Qt version as Appveyor

vadi2 committed:
Build in release mode

vadi2 committed:
Update order of installs to reflect Appveyors

vadi2 committed:
Handle 32bit

vadi2 committed:
Minor cleanup

vadi2 committed:
Undo upload change

vadi2 committed:
Try building with RelWithDebInfo configuration

vadi2 committed:
Minor cleanup

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

vadi2 committed:
Debug - print every incoming line

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

vadi2 committed:
Add backward-cpp

vadi2 committed:
Plug in backward-cpp

vadi2 committed:
Revert "Add backward-cpp"

This reverts commit 85b33ceb0126633d9fa88e37dddb4be9f92b9694.

vadi2 committed:
Revert "Plug in backward-cpp"

This reverts commit 860783fe4f62a356b37a0a689d5cb739a5c4a85f.

vadi2 committed:
Add tracing calls

vadi2 committed:
Fix compile

vadi2 committed:
Fix TLuaInterpreter as well

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

vadi2 committed:
Revert "Revert "Plug in backward-cpp""

This reverts commit a85f3eaeebe65d45ba55badc6fda79f3d51b549a.

vadi2 committed:
Revert "Revert "Add backward-cpp""

This reverts commit 0eb2817bee11283fbf930db5d3eaa61d35ccc538.

vadi2 committed:
Sprinkle qDebug()s in parseJSON

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

# Conflicts:
#	src/CMakeLists.txt

vadi2 committed:
Add debugs to callEventHandler

vadi2 committed:
Add more debug

vadi2 committed:
Wrap all lua_pcalls in debug

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

vadi2 committed:
Show when compiling Lua

vadi2 committed:
Remove stray 'exec'

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

vadi2 committed:
Back to installing Lua

vadi2 committed:
Try running built Lua

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

# Conflicts:
#	src/TConsole.cpp

vadi2 committed:
Fix Lua check to only run on Windows

vadi2 committed:
Check Lua before building Mudlet

vadi2 committed:
Check what we built

vadi2 committed:
Update Lua port to latest as vcpkg has it

vadi2 committed:
Debug Lua install logs

vadi2 committed:
Cache Qt install

vadi2 committed:
Fix cat paths

vadi2 committed:
Copy liblua to lua.exe's location

vadi2 committed:
Copy liblua.dll and lua.exe to where Luarocks can find them

vadi2 committed:
Fix comment

vadi2 committed:
Create path before copying into it

vadi2 committed:
Copy Lua link library as well

vadi2 committed:
Improve comment

vadi2 committed:
Don't create the lib too early

vadi2 committed:
Copy in Lua header as well

vadi2 committed:
Set InstallVcpkgLua

vadi2 committed:
Debug installing Lua

vadi2 committed:
Disable installing lua.hpp

vadi2 committed:
Copy lua.hpp ourselves

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

keneanung commented:
/refresh links

keneanung commented:
/refresh links

keneanung commented:
/refresh links

vadi2 committed:
Merge remote-tracking branch 'SlySven/development' into add-windows-packaging

# Conflicts:
#	.github/workflows/build-mudlet.yml

vadi2 committed:
Update to vcpkg which should be able to install zziplib

vadi2 committed:
Install vcpkg zziplib

vadi2 committed:
Cleanup Lua and Yajl commands

vadi2 committed:
Try with latest vcpkg fe5852514a5cf82a3fadec6b9458caf134c451f1

vadi2 committed:
Try installing zziplib first

vadi2 committed:
Install zziplib only

vadi2 committed:
Add zlib log

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

vadi2 committed:
test

vadi2 committed:
Latest vcpkg

vadi2 committed:
Quote for cat

vadi2 committed:
Revert "Latest vcpkg"

This reverts commit c2624246dbee802676da18a9d9c01b47303374b9.

vadi2 committed:
Revert "Quote for cat"

This reverts commit 975d41381f744c656e3a6a565dbc3874a7220c32.

vadi2 committed:
Revert "Merge remote-tracking branch 'origin/development' into add-windows-packaging"

This reverts commit 8dd03c476c94d316a235afe2f13254b27a80948c, reversing
changes made to e36b616d4fb2137a69e0ed901ca7a1f0f5ef73d7.

vadi2 committed:
Revert "Add zlib log"

This reverts commit e36b616d4fb2137a69e0ed901ca7a1f0f5ef73d7.

vadi2 committed:
Revert "test"

This reverts commit a3b64cf9afadc409ef00d5afe0295535451e2c6b.

vadi2 committed:
Add zlib log

vadi2 committed:
Search for mingw32-windres.exe

vadi2 committed:
Use right branch

vadi2 committed:
Don't search

vadi2 committed:
Show zziplib logs

vadi2 committed:
Submit Github Actions-made builds to make.mudlet.org (#4487)

Kebap committed:
Fix typos in documentation & error messages (#4504)

Kebap committed:
Add permGroup(name, "key", parent) (#3151)

* Add permGroup(name, "key", parent)

* set folder key as -1

* Re-set edbee to what it should be

* another end for the group_creation_functions

* allow creating permanent key folders

* Folders start as inactive

Co-authored-by: Vadim Peretokin <vperetokin@gmail.com>

vadi2 committed:
Fix Valgrind complaint around autocompletion functions (#4517)

SlySven committed:
Revise: suggestion to put default mouse scroll to 3 lines (#4509)

That is 3 lines per default (15 degree) click of the (vertical) scroll
wheel. To get the single line per click it can now be obtained by using the
shift key, and that overrides the other override of half a screen height of
lines that the control key gives.

This might close the resurrected #4437.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct QRegularExpression warnings on profile save (#4519)

This was causing messages of the form:
"QRegularExpressionPrivate::doMatch(): called on an invalid
QRegularExpression object" whenever a colour trigger was saved. It was due
to an error in the regular expression defined in:
QStringList XMLexport::remapAnsiToColorNumber(const QStringList&,
                                                         const QList<int>&)

Also remove unneeded #include <QRegularExpression>  in classes that do
not need it.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

Delwing committed:
Allow loading some modules before scripts (#4508)

* allow loading some modules before scripts

* add negative priority remark in infobox of modules

Co-authored-by: Piotr Wilczynski <piotr.wilczynski@bisnode.com>

Delwing committed:
Validate permKey function argument before actually creating permKey (#4513)

* validate permKey function argument before actually create permKey

* correct comment over internal validLuaCode

* Update src/TLuaInterpreter.cpp

Co-authored-by: Vadim Peretokin <vperetokin@gmail.com>

* auto dereference validation results

* move initializer to if brackets

* correct lua validation error message

Co-authored-by: Piotr Wilczynski <piotr.wilczynski@bisnode.com>
Co-authored-by: Vadim Peretokin <vperetokin@gmail.com>

vadi2 committed:
Fix zziplib install

vadi2 committed:
Show zlib build failure log

vadi2 committed:
Update to latest code

vadi2 committed:
Reinstate vcpkg dependencies

vadi2 committed:
Add missing wrapper file

vadi2 committed:
Don't copy lua.hpp

vadi2 committed:
Remove unneeded lua.hpp

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-windows-packaging

# Conflicts:
#	.github/workflows/build-mudlet.yml
#	CI/after_success.ps1
#	src/TLuaInterpreter.cpp
#	src/TLuaInterpreter.h
#	src/mudlet.cpp
#	translations/mudlet.ts

vadi2 committed:
Set icon correctly

vadi2 committed:
Trim trailing spaces

vadi2 committed:
Remove unneeded file

vadi2 committed:
Cleanup

vadi2 commented:
Closing for now, will re-open later.
